scripts/inject-deterministic.py

```diff
@@ -1,17 +1,107 @@
+        string if the CSP header's script source does not contain a nonce.
+        For more background information on CSP and nonce, please refer to
+        https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/
+        Content-Security-Policy/script-src
+        https://developers.google.com/web/fundamentals/security/csp/
+        """
+        for source in self.get_csp_script_sources(headers) or []:
+            if source.startswith("'nonce-"):
+                return source.partition("'nonce-")[-1][:-1]
+        return None
+
+    def get_script_with_nonche(self, script, nonce=None):
+        """
+         Given a nonce, get_script_with_nonche returns the injected script text with the nonce.
+        If nonce is an empty string, getScriptWithNonce returns the script block
+        without attaching a nonce attribute.
+        Some responses may specify a nonce inside their Content-Security-Policy,
+        script-src directive.
+        The script injector needs to set the injected script's nonce attribute to
+        open execute permission for the injected script.
+        """
+        if nonce:
+            return '<script nonce="{}">{}</script>'.format(nonce, script)
+        return '<script>{}</script>'.format(script)
+
+    def get_new_csp_script_diretives(self, headers, sha256):
+        """
+        Update the csp_script_diretives with appropiate information
+        Without this permission a page with a
+        restrictive CSP will not execute injected scripts.
+        For more background information on CSP, please refer to
+        https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/
+        Content-Security-Policy/script-src
+        https://developers.google.com/web/fundamentals/security/csp/
+        """
+        if self.get_csp_script_sources(headers):
+            new_directive = ""
+            add_unsafe = True
+            for token in self.get_csp_script_sources(headers):
+                if token == "'unsafe-inline'":
+                    add_unsafe = False
+                    ctx.log.info("Has unsafe-inline")
+
+                if token.startswith("'sha256-") or \
+                        token.startswith("'sha384-") or \
+                        token.startswith("'sha512-"):
+                    new_directive = " 'sha256-{}'".format(sha256)
+                    ctx.log.info("add directive %s" % new_directive)
+                    add_unsafe = False
+
+            if add_unsafe:
+                ctx.log.info("Add unsafe")
+                new_directive += "'unsafe-inline'"
+
+            return "{} {}".format(new_directive, " ".join(self.get_csp_script_sources(headers)))
+
+    def get_new_csp_header(self, headers, new_csp_script_directives):
+        """
+        get_new_csp_header genetares a new header object contining the updated elements from new_csp_script_directives
+
+        """
+        new_csp_directives = []
+        if new_csp_script_directives:
```

Comment:
We should only be calling this when we have script source directive, so we can remove the `if` block.